homeassistant/components/plum_lightpad.py

```diff
@@ -0,0 +1,55 @@
+"""
+Support for Plum Lightpad switches.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/plum_lightpad
+"""
+import logging
+
+import voluptuous as vol
+
+from homeassistant.const import (
+    CONF_PASSWORD, CONF_USERNAME, EVENT_HOMEASSISTANT_STOP)
+from homeassistant.core import callback
+from homeassistant.helpers import discovery
+import homeassistant.helpers.config_validation as cv
+
+REQUIREMENTS = ['plumlightpad==0.0.9']
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'plum_lightpad'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_USERNAME): cv.string,
+        vol.Required(CONF_PASSWORD): cv.string,
+    }),
+}, extra=vol.ALLOW_EXTRA)
+
+PLUM_DATA = 'plum'
+
+
+def setup(hass, config):
+    """Setup the Plum Lightpad component."""
+    from plumlightpad import Plum
+
+    conf = config[DOMAIN]
+    plum = Plum(conf[CONF_USERNAME], conf[CONF_PASSWORD])
```

Comment:
The plum library uses aiohttp for the cloud connection. We want to use the same aiohttp session for all parts of the home assistant app. Please update the library to accept a session as a keyword argument that is passed to the cloud api client instance.

Get the home assistant aiohttp session via the aiohttp client helper.